### PR TITLE
feat: add movie search service

### LINKED ISSUE #183: Movie API
Movie appears to be top searched along with Weather. 

Create an API based on https://www.themoviedb.org/documentation/api or https://developer.imdb.com/

http://www.omdbapi.com/

### PR DESCRIPTION
**Pull request by h1z3y3**
close #183

powered by: https://developers.themoviedb.org/3/search/search-movies

call service: 

``` bash 
micro call movie Movie.SearchMovie '{"query": "Venom", "primary_release_year": 2021}' 
```

### CODE REVIEW

File: movie/proto/movie.proto [hunk 1]
```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package movie;
+
+option go_package = "./proto;movie";
+
+service Movie {
+  rpc SearchMovie(SearchMovieRequest) returns (SearchMovieResponse) {}
```
Comment (asim) [inline movie/proto/movie.proto:8]: Please rename go Search rather than MovieSearch

File: movie/proto/movie.proto [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+message SearchMovieRequest {
+  // a ISO 639-1 value to display translated data
+  string language = 1;
+  // a text query to search
+  string query = 2;
+  // page to query
+  int32 page = 3;
+  // whether to include adult (pornography) content in the results
+  bool include_adult = 4;
```
Comment (asim) [inline movie/proto/movie.proto:36]: Please remove this flag. We will exclude all pornographic content.

File: movie/publicapi.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "movie",
+  "icon": "🎬",
+  "category": "search",
+  "display_name": "MovieSearch"
```
Comment (asim) [inline movie/publicapi.json:5]: Please rename display name to Movies

File: movie/examples.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
+{
```
Comment (asim) [inline movie/examples.json:1]: Please see another service's example.json file for a proper example. It must structured in a specific way to be used.

File: movie/examples.json [hunk 5]
```diff
@@ -1,66 +1,52 @@
 {
-  "total_results": 3,
-  "total_pages": 1,
-  "page": 1,
-  "results": [
+  "search": [
```
Comment (asim) [inline movie/examples.json:2]: https://github.com/micro/services/blob/master/youtube/examples.json

Example format

```
{
  "search": [{
        "title": "Search for movies",
        "run_check": false,
        "description": "Search for movies",
        "request": {
            // request query
        },
        "response": {
           /// results
        }
   }]
}

**Comment by asim** [comment]
Thank you for the contribution, this API will appear at https://m3o.com/movie shortly.

**Comment by h1z3y3** [comment]
Cool !!! And I noticed that there is a field `run_check`, so what's the difference if the it's `true`.

**Comment by asim** [comment]
If run_check is true then we periodically run the example to check if the API is functioning correctly. When it fails we receive an alert. Otherwise the check is disabled.

**Comment by h1z3y3** [comment]
Ok, I think it's better to set it always be `true`. 😂

**Comment by asim** [comment]
Some APIs are generally quite glitchy or timeout so we don't test anything. In the long term yes that's something we want to do.

Do you have an account on the m3o platform?

**Comment by h1z3y3** [comment]
I just signed up one with my google account: `gmzhaoyang.at.gmail.com`
 
I'm learning about go-micro and micro, so I want to do more contributions on these projects.
So if there is anything I can help, please @me let me know. I'd be pleasure with it.

**Comment by h1z3y3** [comment]
Oh, it's here: https://m3o.com/movie, 
I'm glad to see it getting online.

**Comment by asim** [comment]
We've credited your account with $10. Thanks for your contribution!

**Comment by h1z3y3** [comment]
Wow, thank you very much.

I get an idea about that I can create some [Alfred](https://www.alfredapp.com/) workflows with m3o.com APIs due to the APIs are very suitable for Alfred workflows.
I don't know if it can help to get more users, but I would try it.

**Comment by h1z3y3** [comment]
If I do more contribution, could I apply to join [Micro Organization](https://github.com/micro) to make my resume shine?

**Comment by asim** [comment]
We welcome more contribution of APIs and will continue to provide credit for those contributions.

We are not currently adding members to the micro org but may define some sort of "creator" programme which you can then identify yourself as a Micro Creator or similar in the future.

**Comment by h1z3y3** [comment]
🆒

**Pull request merged**